lib/grp.gi

```diff
@@ -573,6 +577,32 @@ InstallImmediateMethod( IsSolvableGroup, IsGroup and HasSize, 10,
     TryNextMethod();
     end );
 
+InstallMethod( IsSolvableGroup,
+    "if group size is known and is 2m or p^a q^b",
+    [ IsGroup and HasSize ], 25,
+    function( G )
+    local size;
+    size := Size( G );
+    if IsInt( size ) then
+      if size mod 4 <> 0 then
```

Comment:
In view of the immediate metjod just above: Will this case ever arise?

Reply:
As long as the Immediate method is in place, it won't. 

However, I would still prefer this to be left here, in case the immediate method is ever removed (in which case this method would not need adjusting). 

Originally, I was thinking about changing the immediate method, but finally decided against it, because felt that a possible check for being prime power, even though it is reasonably cheap, is not as cheap as a modulo 4 check is.